app/models/predefined_user_parameter.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class PredefinedUserParameter < ApplicationRecord
+  belongs_to :user
+  belongs_to :intervention
+  belongs_to :health_clinic, optional: true
+
+  validates :slug, uniqueness: true
+
+  before_validation :generate_slug
+
+  def generate_slug(suggested_slug = nil)
```

Comment:
I believe it's the third occurrence of this recursive slug generation, do you think it would be beneficial to put this logic in one place? 🔥😅

Reply:
Only second XD